packages/build-info/src/frameworks/gatsby.ts

```diff
@@ -0,0 +1,53 @@
+import { gte } from 'semver'
+
+import { BaseFramework, Category, Framework } from './framework.js'
+
+export class Gatsby extends BaseFramework implements Framework {
+  id = 'gatsby'
+  name = 'Gatsby'
+  configFiles = ['gatsby-config.js']
+  npmDependencies = ['gatsby']
+  category = Category.SSG
+  staticAssetsDirectory = 'static'
+
+  dev = {
+    command: 'gatsby develop',
+    port: 8000,
+    pollingStrategies: [{ name: 'TCP' }, { name: 'HTTP' }],
+  }
+
+  build = {
+    command: 'gatsby build',
+    directory: 'public',
+  }
+
+  logo = {
+    default: '/logos/gatsby/default.svg',
+    light: '/logos/gatsby/light.svg',
+    dark: '/logos/gatsby/dark.svg',
+  }
+
+  env = {
+    GATSBY_LOGGER: 'yurnalist',
+    GATSBY_PRECOMPILE_DEVELOP_FUNCTIONS: 'true',
+    AWS_LAMBDA_JS_RUNTIME: 'nodejs14.x',
+    NODE_VERSION: '14',
+  }
+
+  async detect()
+  async detect(): Promise<this | undefined> {
+    const detected = await super.detect()
+    if (detected) {
+      if (this.version && gte(this.version, '5.0.0')) {
+        this.env.NODE_VERSION = '18'
+        this.env.AWS_LAMBDA_JS_RUNTIME = 'nodejs18.x'
+      }
```

Comment:
@danez is this the fix you need for https://github.com/netlify/pod-compute/issues/309

cc @minivan

Reply:
yes! that would be what we need

Reply:
These variables are only used in the cli afaik. So for the build system we still need to figure out a way to set it.

Reply:
@danez could you elaborate on that? What would be needed to set, or how would the issue be fixed in the best way for you?

Happy to hop on a call as well :)